scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,329 @@
+
+        let (begin, end) = match matches_by_name.as_slice() {
+            [single_match] => Ok((start, (*single_match).clone())),
+            [] => Err(anyhow!("Couldn't find a matching end anchor for {start:?}")),
+            multiple_ends => Err(anyhow!("Found too many matching anchor ends for anchor: {start:?}. The matching ends are: {multiple_ends:?}")),
+        }?;
+
+        match check_validity_of_anchor_pair(&begin, &end) {
+            None => Ok((begin, end)),
+            Some(err) => {
+                let err_msg = err.to_string();
+                Err(anyhow!("{err_msg}"))
+            }
+        }
+    }).partition_result();
+
+    let additional_errors = filter_unused_ends(&ends, &pairs)
+        .into_iter()
+        .map(|unused_end| anyhow!("Missing anchor start for {unused_end:?}"))
+        .collect::<Vec<_>>();
+
+    let start_only = pairs.into_iter().map(|(begin, _)| begin).collect();
+
+    (start_only, chain!(errors, additional_errors).collect())
+}
+
+fn filter_unused_ends<'a>(ends: &'a [Anchor], pairs: &[(Anchor, Anchor)]) -> Vec<&'a Anchor> {
+    ends.iter()
+        .filter(|end| {
+            let end_used_in_pairs = pairs.iter().any(|(_, used_end)| *end == used_end);
+            !end_used_in_pairs
+        })
+        .collect()
+}
+
+fn check_validity_of_anchor_pair(begin: &Anchor, end: &Anchor) -> Option<anyhow::Error> {
+    if begin.line_no > end.line_no {
+        Some(anyhow!("The end of the anchor appears before the beginning. End anchor: {end:?}. Begin anchor: {begin:?}"))
+    } else {
+        None
+    }
+}
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+struct Anchor {
+    line_no: usize,
+    name: String,
+    file: PathBuf,
+}
+
+#[allow(dead_code)]
+enum TestEnum {
+    Anchor(Vec<Anchor>),
+    Include(Vec<Include>),
+    Errors(Vec<Error>),
+}
+
+fn extract_starts_and_ends(
+    text_w_anchors: &str,
+) -> anyhow::Result<(Vec<Anchor>, Vec<Anchor>), Error> {
+    let apply_regex = |regex: Regex| {
+        text_w_anchors
+            .lines()
+            .filter_map(|line| regex.captures(line))
+            .map(|capture| {
+                let file = PathBuf::from(&capture[1]).canonicalize()?;
+                let line_no = &capture[2];
+                let anchor_name = &capture[3];
+
+                Ok(Anchor {
+                    line_no: line_no.parse().unwrap(),
+                    name: anchor_name.to_string(),
+                    file,
+                })
+            })
+            .collect::<Result<Vec<_>, Error>>()
+    };
+
+    let begins = apply_regex(Regex::new(
+        r"^(.+):(\d+):\s*(?:/{2,}|/\*)\s*ANCHOR\s*:\s*([\w_-]+)\s*(?:\*/)?",
+    )?)?;
+    let ends = apply_regex(Regex::new(
+        r"^(.+):(\d+):\s*(?:/{2,}|/\*)\s*ANCHOR_END\s*:\s*([\w_-]+)\s*(?:\*/)?",
+    )?)?;
+
+    Ok((begins, ends))
+}
+
+fn search_for_patterns_in_project(pattern: &str) -> anyhow::Result<String> {
+    let grep_project = std::process::Command::new("grep")
+        .arg("-I")
```

Comment:
You can use longer versions of these flags for readability. '-IHRn' is only readable to sys admins.